src/main/java/duke/main/Duke.java

```diff
@@ -0,0 +1,204 @@
+            String inputFromUser = sc.nextLine();
+            Instruction instruction = duke.getInstr(inputFromUser);
+        
+            switch (instruction) {
+            case LIST:
+                duke.ui.listing(duke.list);
+                break;
+            case MARK:
+                duke.ui.handleMarkError(inputFromUser, duke.list);
+                Task taskToBeMarked = duke.parser.getTaskTobeMarked(inputFromUser, duke.list);
+                taskToBeMarked.markDone();
+                duke.ui.marking(taskToBeMarked);
+                try {
+                    duke.storage.changeFileContent(duke.list);
+                } catch (IOException e) {
+                    duke.ui.changingFileError();
+                }
+                break;
+            case UNMARK:
+                duke.ui.handleUnmarkError(inputFromUser, duke.list);
+                Task taskToBeUnmarked = duke.parser.getTaskToBeUnmarked(inputFromUser, duke.list);
+                taskToBeUnmarked.markUndone();
+                duke.ui.unmarking(taskToBeUnmarked);
+                try {
+                    duke.storage.changeFileContent(duke.list);
+                } catch (IOException e) {
+                    duke.ui.changingFileError();
+                }
+                break;
+            case TODO:
+                duke.ui.handleTodoError(inputFromUser);
+                Task todoTask = duke.parser.createToDo(inputFromUser);
+                duke.ui.echo(todoTask, duke.list);
+                try {
+                    duke.storage.addTaskToFile((todoTask));
+                } catch (IOException e) {
+                    duke.ui.changingFileError();
+                }
+                break;
+            case DEADLINE:
+                duke.ui.handleDeadlineError(inputFromUser);
+                Task deadlineTask = duke.parser.createDeadline(inputFromUser, duke.ui);
+                duke.ui.echo(deadlineTask, duke.list);
+                try {
+                    duke.storage.addTaskToFile((deadlineTask));
+                } catch (IOException e) {
+                    duke.ui.changingFileError();
+                }
+                break;
+            case EVENT:
+                duke.ui.handleEventError(inputFromUser);
+                Task eventTask = duke.parser.createEvent(inputFromUser, duke.ui);
+                duke.ui.echo(eventTask, duke.list);
+                try {
+                    duke.storage.addTaskToFile((eventTask));
+                } catch (IOException e) {
+                    duke.ui.changingFileError();
+                }
+                break;
+            case DELETE:
+                int indexOfTaskToDelete = Integer.parseInt(inputFromUser.substring(7));
+                duke.ui.handleDeleteError(duke.list,indexOfTaskToDelete);
+                Task taskToDelete = duke.parser.getTaskToDelete(inputFromUser, duke.list);
+                duke.ui.deleting(taskToDelete, duke.list);
+                duke.list.delete(taskToDelete);
```

Comment:
this could be "deleteTask" just to be a little more specific especially when put in close proximity with your ui method "deleting"